### PR TITLE
adding changes for http proxy

### PR DESCRIPTION
**Pull request by joytandon17**
Changes to add http proxy

### CODE REVIEW

File: main/createUiDefinition.json [hunk 1]
```diff
@@ -69,6 +69,18 @@
           "validationMessage": "Insert valid Jointoken."
         },
         "visible": "true"
+      },
+      {
+        "name": "accessproxy",
+        "type": "Microsoft.Common.TextBox",
+        "label": "http proxy",
+        "toolTip": "Pass Accessproxy.",
```
Comment (nnallamo) [inline main/createUiDefinition.json:77]: Can we change these tool tip description . we can use text from bootstrap ui for reference

File: main/createUiDefinition.json [hunk 2]
```diff
@@ -69,6 +69,18 @@
...
+        "constraints": {
+          "required": false,
+          "regex": "(http|https)?:\\/\\/(\\S+)",
+          "validationMessage": "Pass correct Access Proxy"
```
Comment (jkraj) [inline main/createUiDefinition.json:81]:
```suggestion
          "validationMessage": "Enter correct HTTP Proxy"
```

File: main/createUiDefinition.json [hunk 3]
```diff
@@ -69,6 +69,18 @@
...
         "visible": "true"
+      },
+      {
+        "name": "accessproxy",
+        "type": "Microsoft.Common.TextBox",
+        "label": "http proxy",
+        "toolTip": "Pass Accessproxy.",
+        "constraints": {
+          "required": false,
+          "regex": "(http|https)?:\\/\\/(\\S+)",
```
Comment (jkraj) [inline main/createUiDefinition.json:80]: does this allow `http://xyz.com:8080` ?

File: main/createUiDefinition.json [hunk 4]
```diff
@@ -186,6 +198,7 @@
     ],
     "outputs": {
       "jointoken": "[basics('jointoken')]",
+      "accessproxy": "[basics('accessproxy')]",
```
Comment (jkraj) [inline main/createUiDefinition.json:201]:
```suggestion
      "accessproxy": "[basics('httpproxy')]",
```
Except here everywhere name is `httpproxy`, use `accesproxy` only inside cloud-init user data

File: main/createUiDefinition.json [hunk 5]
```diff
@@ -69,6 +69,18 @@
           "validationMessage": "Insert valid Jointoken."
         },
         "visible": "true"
+      },
+      {
+        "name": "accessproxy",
+        "type": "Microsoft.Common.TextBox",
+        "label": "http proxy",
```
Comment (jkraj) [inline main/createUiDefinition.json:76]:
```suggestion
        "label": "HTTP Proxy",
```

File: main/createUiDefinition.json [hunk 6]
```diff
@@ -69,6 +69,18 @@
           "validationMessage": "Insert valid Jointoken."
         },
         "visible": "true"
+      },
+      {
+        "name": "accessproxy",
```
Comment (jkraj) [inline main/createUiDefinition.json:74]:
```suggestion
        "name": "httpproxy",
```

File: main/createUiDefinition.json (hunk 2 above)
Comment (joytandon17) [reply main/createUiDefinition.json]: done

File: main/createUiDefinition.json (hunk 3 above)
Comment (joytandon17) [reply main/createUiDefinition.json:80]: yes this checks whether address starts from http or https and does not have a whitespace in between

File: main/createUiDefinition.json (hunk 4 above)
Comment (joytandon17) [reply main/createUiDefinition.json]: done

File: main/createUiDefinition.json (hunk 5 above)
Comment (joytandon17) [reply main/createUiDefinition.json]: done

File: main/createUiDefinition.json (hunk 6 above)
Comment (joytandon17) [reply main/createUiDefinition.json]: done

**Review by andrei-maroz — CHANGES_REQUESTED** [review]

**Review by andrei-maroz — APPROVED** [review]

**Review by nnallamo — APPROVED** [review]

**Review by jkraj — APPROVED** [review]

**Pull request merged**